### PR TITLE
ZFIN-9294: Error message for adding fluorescent protein

### PR DESCRIPTION
**Pull request by rtaylorzfin**
Show a more informative error message when a fluorescent protein isn't in our DB yet:

Screenshot 2024-07-02 at 2 37 43 PM

### CODE REVIEW

File: source/org/zfin/fish/presentation/EfgAPIController.java [hunk 1]
```diff
@@ -9,45 +9,64 @@
...
 
 @RestController
 @RequestMapping("/api/efg")
 public class EfgAPIController {
 
+    //Create nested structure for errors such that js can call: responseJSON.fieldErrors[0].message (for example)
     @JsonView(View.API.class)
-    public record Fpbase(String fpId, String abbreviation) {
-    }
+    private record FpbaseDTO(String fpId, String abbreviation, FpbaseResponseFieldError[] fieldErrors) {
```
Comment (cmpich) [inline source/org/zfin/fish/presentation/EfgAPIController.java:24]: Why did you change the name to FpbaseDTO? To indicate it is not a business object? I wonder if we should have a naming convention for record's.

Comment (rtaylorzfin) [reply source/org/zfin/fish/presentation/EfgAPIController.java:24]: I tend to think of these value objects as being pretty much the same as DTO objects.  Its purpose is to represent the fpbase entity for transmitting for the server to the client--so that seems to fit the definition of DTO to me. If you prefer a different name, that's fine with me.

Comment (cmpich) [reply source/org/zfin/fish/presentation/EfgAPIController.java:24]: So you want to make it clear that this object only serves the purpose of data transfer and thus make it clear in the class name? I can agree to that.

**Review by cmpich — APPROVED** [review]
Looks good.
